src/celengine/shadermanager.cpp

```diff
@@ -2431,19 +2415,19 @@ ShaderManager::buildFragmentShader(const ShaderProperties& props)
             }
 
             if ((props.texUsage & ShaderProperties::NightTexture) != 0)
-                source += "totalLight += l * " + FragLightProperty(i, "brightness") + ";\n";
+                source += "totalLight += l * " + LightProperty(i, "brightness") + ";\n";
 
             if (props.hasShadowsForLight(i))
                 source += ShadowsForLightSource(props, i);
 
             std::string illum(props.hasShadowsForLight(i) ? "l * shadow" : "l");
-            source += "diff.rgb += " + illum + " * " + FragLightProperty(i, "color") + ";\n";
+            source += "diff.rgb += " + illum + " * " + LightProperty(i, "diffuse") + ";\n";
```

Comment:
why do you replace color with diffuse?

Reply:
Because this replicates the same behavior as the original version. The previous code used both `fragLightColor[i]` and `lights[i].diffuse` to store the quantity `lightColor.cwiseProduct(diffuseColor)`

See L3672/3681 at https://github.com/CelestiaProject/Celestia/pull/1859/files#diff-69cd8d8749fa1d3212a10624731fe412e2afaaed33931085c3427a3009a981acL3667-L3682

There was not originally a `color` property on the lights, this is a new addition to provide the unchanged light color to the atmosphere shader.